### ISSUE TITLE
Styles: Links and button outline on focus is not accessible

### ISSUE THREAD

**Issue by kkoryaka** [cagov/design-system#755]
According to WCAG 1.4.11 the outline color of the interactive element on focus should have 3:1 color contrast ratio with adjacent colors. 

Current outline color: # fbad23
Contrast ratio with white background: 1.89

Screen shot example:
![Screen Shot 2022-03-22 at 11 07 17 AM](https://user-images.githubusercontent.com/31669748/159547316-f8910959-6c7d-40ba-b667-66392739be8a.png)

Also, notice that outline color on dark backgrounds should also have 3:1 color ratio. That means that outline color might be different for dark backgrounds.

**Comment by jayliquek** [comment]
Hello! apologies for intruding like this, I left an issue but not sure if anyone of the contributors saw. Would love to help out with this ticket if possible!

**Comment by kmbrlygl** [comment]
Hey @jayliquek thanks for reaching out and offering support! I pinged the team so they saw your message. We'll be getting back to you soon.

**Comment by jayliquek** [comment]
Hey! Apologies for the MIA. I've been reading about WCAG and playing around with color explorations. 

## Button outline 
One thing to consider if there might be a dark mode in the future is the background used for dark mode. 
Although the outline color I've suggested meets the requirements, the color of the button no longer meets the 3:1 contrast ratio with the black background. However, changing the color of the button could also break with consistency of the design system. 
<img width="900" alt="Button Color Explorations" src="https://user-images.githubusercontent.com/45698971/164993489-5391a511-99f6-469c-96e8-9f1a22db7d1e.png">

## Links outline
I've checked against the contrast color checker and I think accent-2-700 could be worth considering for light backgrounds and accent 2-300 could be used for dark backgrounds. I'm not super sure about the text color pairings, but have considered white, and primary-100. 
<img width="528" alt="Screen Shot 2022-04-24 at 3 39 56 PM" src="https://user-images.githubusercontent.com/45698971/164993590-5dccf972-4c1d-4396-98ef-1ddb8969460d.png">

You can find the figma link [here](https://www.figma.com/file/Y6cmnT9CE8IoaNV6FiZWcE/Design-System-Color-Share-(Copy)?node-id=0%3A1)

Would love to hear your thoughts @kmbrlygl ! If you have moved on would love to see what you guys ended up deciding.

**Comment by sharon-rabang** [comment]
Hi @jayliquek, Thanks so much for sharing these color explorations for consideration! We actually started doing some similar explorations already and are well on our way. We can take this one from here and definitely share what we end up with when it's ready.

**Comment by sharon-rabang** [comment]
In Slack: cagov-design-system-design

Hello! Update on [#700](https://github.com/cagov/design-system/issues/700) Design focus states (related to [#755](https://github.com/cagov/design-system/issues/755) Links and button outline on focus is not accessible)

Thanks for all of the helpful feedback at our 4/12/22 weekly design meeting and from today's mini meeting. I did more studies and found a single color that works on both our light and dark backgrounds. 

Here's the direction I propose moving forward with in [Figma](https://www.figma.com/file/4kfjgYBUbX6gi2yQZaBCwy/Focus-states?node-id=8%3A143):
	• Custom focus style: Box-shadow (has the widest range of aesthetic design capabilities)
	• Focus color: 
		○ 4px #2491FF on light, #FFFFFF
Contrast ratio 3.2:1 
		○ 4px #2491FF on dark, #003688
Contrast ratio 3.5:1
		
Let me know if you have any feedback, before moving forward with a developer.

Thanks!

**Comment by yesenia-gtz** [comment]
@zakiya  this one is ready to development.

**Comment by xjensen** [comment]
I have some concerns about this request. I'm not sure it's ready.

In the Design System code, we do not hard-code specific colors. We're relying on our design tokens for colors throughout (`accent1-300`, `primary-900`, etc.). This is true of focus states too. Tokens are important because we're already using the Design System across multiple sites with different color schemes. Keeping our colors in tokens ensures we don't have an out-of-place pop of blue on drought.ca.gov, for example. 

I think it's important that we communicate color changes in terms of these design tokens. If we need to establish new tokens, sure, but we should be explicit about that.

The other concern I have here is with respect to using `box-shadow` to denote application state. Box-shadow is often hidden when using high-contrast color modes within the operating system. Therefore, high-contrast users would miss important visual queues. It's an accessibility problem. We do have workarounds in recent browsers (like `@media (forced-colors: active)`), but we'll still need to have a fallback to `outline`. It's a bit messy, and something that needs to be noted if we move forward.

**Comment by yesenia-gtz** [comment]
Good point, let's table this for now as we think through now to address the a11y concerns.

**Comment by xjensen** [comment]
Removing my assignment to clean up my list. Feel free to reassign to me when ready.
